### DISCUSSION TITLE
Grafana

### DISCUSSION THREAD

**Discussion by KurtNeven** [WLAN-Pi/feedback#62] [WLAN Pi OS] [answered]
Hello,
Just received the WlanPi M4 and still starting to get to know the product.
My question is related to the Grafana portion. 

Is it normal that the librespeed test takes so long ? Also plotting the data in the graphs is slow I get the messages "data outside time range).

My timewindow is on Last 5 minutes and the refresh on 5 seconds. 

Am I doing something wrong somewhere ? Or is the usecase of this test different than I thought is was ? RealTime monitoring of internet connection.

Kind regards,
Kurt

**Comment by jiribrejcha** [comment, marked as answer]
**Q: Is it normal that the librespeed test takes so long ?**
A: The test itself runs about 20-30 seconds. And it is executed every 5 minutes (300 seconds). Expect samples every 5 minutes. You can easily change this interval in `/opt/wlanpi-grafana/internet/stream-internet.sh`. After you change the interval, make sure you restart the Data Stream.

**Q: Also plotting the data in the graphs is slow I get the messages "data outside time range).**
A: That might happen when the Pi doesn't have correct time set (perhaps NTP wasn't available). You can simply change the view and see the past 15 minutes for example. 

**Q: Am I doing something wrong somewhere ? Or is the usecase of this test different than I thought is was ? RealTime monitoring of internet connection.**
A: It checks internet connection every 5 minutes. By changing the interval above to test connection more frequently. Just make sure the test finishes before next test starts. Otherwise you might end up running more speed tests at the same time which will impact the results.

**Comment by KurtNeven** [comment]
Thank you for the update and my apologies for replying this late.
I just had to wait a little longer in order to get the first readings. Still learning to use the tool but I like it a lot.


Grts.
Kurt





Kurt Neven
Solutions Architect
T +32 16 38 08 57
M +32 473 23 99 20
***@***.******@***.***>

***@***.***<http://www.quant.be>

**Reply by jiribrejcha** [reply]
No worries, there is no rush 👍🏻 Enjoy using your WLAN Pi!

At the Wireless LAN Professionals Conference (WLPC), we deliver a WLAN Pi Deep Dive with 6 hours of hands-on labs and step by step instructions. It would be great to have you next time.
